### ISSUE TITLE
More details about the first phase of training

### ISSUE THREAD

**Issue by ChenyangZhang00** [noorahmedds/OrCo#1]
Can you share in detail how you completed the first phase of training with solo-learn?

**Comment by noorahmedds** [comment]
I add a small snippet to the supcon.py file inside _solo/methods_ in the training_step function.
```
if self.apply_infonce:
            indexes = batch[0]
            indexes = indexes.repeat(n_augs)
            selfsuploss = simclr_loss_func(
                z,
                indexes=indexes,
                temperature=self.temperature,
            )
            
            nce_loss = ((1-self.simclr_weight) * nce_loss) + (self.simclr_weight*selfsuploss)
```

Along with the correct imports: `from solo.losses.simclr import simclr_loss_func`
I fix the simclr_weight at 0.1 for all datasets.

**Comment by ChenyangZhang00** [comment]
> I add a small snippet to the supcon.py file inside _solo/methods_ in the training_step function.
> 
> ```
> if self.apply_infonce:
>             indexes = batch[0]
>             indexes = indexes.repeat(n_augs)
>             selfsuploss = simclr_loss_func(
>                 z,
>                 indexes=indexes,
>                 temperature=self.temperature,
>             )
>             
>             nce_loss = ((1-self.simclr_weight) * nce_loss) + (self.simclr_weight*selfsuploss)
> ```
> 
> Along with the correct imports: `from solo.losses.simclr import simclr_loss_func` I fix the simclr_weight at 0.1 for all datasets.

Which config did you use to execute the following command?
![image](https://github.com/noorahmedds/OrCo/assets/74999187/1ec47df3-d302-45db-9979-4ba4c1be6a17)

**Comment by noorahmedds** [comment]
I just cloned the pretrain/cifar/supcon.yaml and added the apply_infonce parameter. This should directly work for CIFAR but you would have to ensure that your dataloader picks only the first 60 classes. This is possible in their dataloading script as well.

**Comment by ChenyangZhang00** [comment]
> I just cloned the pretrain/cifar/supcon.yaml and added the apply_infonce parameter. This should directly work for CIFAR but you would have to ensure that your dataloader picks only the first 60 classes. This is possible in their dataloading script as well.

OK, thanks for your reply.

**Comment by ChenyangZhang00** [comment]
> I just cloned the pretrain/cifar/supcon.yaml and added the apply_infonce parameter. This should directly work for CIFAR but you would have to ensure that your dataloader picks only the first 60 classes. This is possible in their dataloading script as well.

Can you share how you set up the dataloader to only pick the first 60 classes?

The following is my modification method. I modified the prepare_dataloader function in solo/data/pretrain_dataloader.py:

    all_labels = train_dataset.targets
    
    labels_list = list(range(60))
    selected_indices = [index for index, label in enumerate(all_labels) if label in labels_list]

    subset_sampler = Subset(train_dataset, selected_indices)
    random_sampler =  RandomSampler(subset_sampler, replacement=True)  
    train_loader = DataLoader(
        subset_sampler,
        batch_size=batch_size,
        shuffle=False,
        num_workers=num_workers,
        pin_memory=True,
        drop_last=True,
        sampler=random_sampler
    )
    return train_loader

**Comment by noorahmedds** [comment]
Seems correct for now. Maybe I can share my fork of the solo-learn repo as well. This should happen within this coming week. Thank you for your patience.

**Comment by ChenyangZhang00** [comment]
> Seems correct for now. Maybe I can share my fork of the solo-learn repo as well. This should happen within this coming week. Thank you for your patience.

That’s great, looking forward to your repo!

**Comment by zjn0613** [comment]
> I just cloned the pretrain/cifar/supcon.yaml and added the apply_infonce parameter. This should directly work for CIFAR but you would have to ensure that your dataloader picks only the first 60 classes. This is possible in their dataloading script as well.

It seems that I can't find the apply_infonce in the repository, as well as the additional snippets mentioned later in solo/methods. Could you kindly share the related parts or let me know what I should do?

**Comment by dongdongswd** [comment]
Hello, I am trying to reproduce the pre-training code now, but I have encountered some problems, could you please share your pre-training code？

**Comment by noorahmedds** [comment]
This got put on the back-burner for a while. I have added a minimal version of solo-learn from my archives to the current project in a sub-dir. This refactoring has unfortunately not been tested, so please make best use of your own programming judgement until I am able to further refactor the pretraining code. Sincere apologies for being so extremely delayed on this.

**Comment by dongdongswd** [comment]
Thanks a lot for sharing the code!
